lib/api/build.rs

```diff
@@ -241,6 +241,10 @@ fn configure_validation(builder: Builder) -> Builder {
             ("DiscoveryQuery.context", ""),
             ("ContextQuery.context", ""),
         ], &[])
+        .field_attribute("DatetimeRange.lt", "#[serde(skip)]")
+        .field_attribute("DatetimeRange.gt", "#[serde(skip)]")
+        .field_attribute("DatetimeRange.lte", "#[serde(skip)]")
+        .field_attribute("DatetimeRange.gte", "#[serde(skip)]")
```

Comment:
Is it not possible to make a serializer for `prost_types::Timestamp`? 

We should also add validation for these types, to make sure they fit within 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z, as established by the [`prost_types::Timestamp` documentation](https://docs.rs/prost-types/latest/prost_types/struct.Timestamp.html#fields)

Reply:
We likely have a more limited range of supported times due to us fitting milliseconds in i64.

Reply:
It's possible through field annotations, and I'll do it. I've just followed the example of `SnapshotDescription.creation_time` which also uses `#[serde(skip)]`.

Reply:
Added validation and serializer using `prost_wkt_types` crate.